tests/installation/grub_test.pm

```diff
@@ -49,12 +49,14 @@ sub handle_installer_medium_bootup {
             assert_screen 'grub2';
         }
         # use firmware boot manager of aarch64 to boot upgraded system
-        if (check_var('ARCH', 'aarch64') && get_var('UPGRADE')) {
+        if (check_var('ARCH', 'aarch64')) {
             send_key_until_needlematch 'inst-bootmenu-boot-harddisk', 'up';
             send_key 'ret';
             $self->handle_uefi_boot_disk_workaround;
         }
         else {
+            # we need to assure that we boot from the disk
+            send_key_until_needlematch 'inst-bootmenu-boot-harddisk', 'up';
```

Comment:
I believe we will need new needles for this behavior. It also seems not to be an issue for x86_64: https://openqa.suse.de/tests/2060951#step/grub_test/1 And to be honest, I guess it's not even arch specific, but rather UEFI, and we have a code  path on L64. Therefore, I'm not sure if we should try to non-existing issues, especially if don't gain clear code path which is same for all.

Reply:
I don't really understand your last sentence but I just assume you want to mention the same as what I did in before: That it does not make any sense to call the same code in both the "if" and "else"-branch :)

Reply:
What do you think about something like this:

```perl
sub handle_installer_medium_bootup {
    my ($self) = @_; 

    return unless (check_var("BOOTFROM", "d") || (get_var('UEFI') && get_var('USBBOOT')));

    assert_screen 'inst-bootmenu';

    if (check_var("BOOTFROM", "d") && check_var("AUTOUPGRADE") && check_var("PATCH")) {
        assert_screen 'grub2';
    }

    send_key_until_needlematch 'inst-bootmenu-boot-harddisk', 'up'; 
    send_key 'ret';

    # use firmware boot manager of aarch64 to boot upgraded system
    if (check_var('ARCH', 'aarch64')) {
        $self->handle_uefi_boot_disk_workaround;
    }
}
```

Reply:
That looks much better, but we need thorough testing, that we don't break anything with it.